watershed_workflow/split_hucs.py

```diff
@@ -225,9 +238,40 @@ def partition(list_of_shapes, abs_tol=1.0, rel_tol=1.e-3):
 
             s2 = s2.buffer(abs_tol)
             if watershed_workflow.utils.intersects(s1, s2):
-                s2 = s2.difference(s1)
-                list_of_shapes[j] = s2.difference(s1)
-
+                inter = s1.intersection(s2)
+                
+                if isinstance(inter, shapely.geometry.Polygon):
+                    logging.debug(f"s1: shapes[{i}], s={s1.area} intersects s2: shapes[{j}], s={s2.area}")
+                    logging.debug(f"the intersected area is: {inter.area}")
+                    if s2.within(s1):
```

Comment:
This shouldn't be possible.  This would mean that s2 was inside of s1 BEFORE we buffered it, which means the user provided invalid input.  Can you provide an example of what you're trying to fix here?  I suspect we should, rather than editing this, ask the user to make the shapes be at least close to a partition.